File: 1.Two-Sum/Two-Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        positions = {}
        
        for idx in range(len(nums)):
            second_num = target - nums[idx]
            if second_num in positions:
                return [positions[second_num], idx]
            
            positions[nums[idx]] = idx
            
        return []

File: 1.Two-Sum/Two-Sum.java
/** 
 * link : https://leetcode.com/problems/two-sum/
 * 
 * Time complexity : O(N), where N is size of the given list of numbers
 * Space cimplexity : O(N), where N is size of the given list of numbers
*/

import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int LEN = nums.length;
        HashMap<Integer, Integer> positions = new HashMap<>();
        
        for(int itr = 0; itr < LEN; itr++){
            if(positions.containsKey(target - nums[itr])){
                return new int[]{positions.get(target - nums[itr]), itr};
            }
            
            positions.put(nums[itr], itr);
        }
        
        return new int[]{};
    }
}